src/main/java/org/opensearch/knn/index/KNNIndexShard.java

```diff
@@ -100,6 +102,32 @@ public void warmup() throws IOException {
         }
     }
 
+    /**
+     * Removes all the k-NN segments for this shard from the cache.
+     * Adding write lock onto the NativeMemoryAllocation of the index that needs to be evicted from cache.
+     * Write lock will be unlocked after the index is evicted. This locking mechanism is used to avoid
+     * conflicts with queries fired on this index when the index is being evicted from cache.
+     */
+    public void clearCache() {
+        String indexName = getIndexName();
+        Optional<NativeMemoryAllocation> indexAllocationOptional;
+        NativeMemoryAllocation indexAllocation;
+        indexAllocationOptional = nativeMemoryCacheManager.getIndexMemoryAllocation(indexName);
+        if (indexAllocationOptional.isPresent()) {
+            indexAllocation = indexAllocationOptional.get();
+            indexAllocation.writeLock();
+            logger.info("[KNN] Evicting index from cache: [{}]", indexName);
+            try (Engine.Searcher searcher = indexShard.acquireSearcher("knn-clear-cache")) {
+                getAllEnginePaths(searcher.getIndexReader()).forEach((key, value) -> nativeMemoryCacheManager.invalidate(key));
+            } catch (IOException ex) {
+                logger.error("[KNN] Failed to evict index from cache: [{}]", indexName);
```

Comment:
I think we need to log exception as well

Reply:
+1 we need to log the exception trace.